mm2src/coins/solana/solana_tests.rs

```diff
@@ -335,3 +337,112 @@ fn solana_test_tx_history() {
     }
     log!("{}", serde_json::to_string(&history).unwrap());
 }
+
+#[test]
+fn solana_coin_send_and_refund_maker_payment() {
+    let passphrase = "federal stay trigger hour exist success game vapor become comfort action phone bright ill target wild nasty crumble dune close rare fabric hen iron".to_string();
+    let (_, coin) = solana_coin_for_test(passphrase, SolanaNet::Devnet);
+    let solana_program_id = "AEwou7VU9bahWRSN2zzgMGqozNY2G27U9fbbdLF9myZx";
+    let solana_program_id = bs58::decode(solana_program_id).into_vec().unwrap_or_else(|e| {
+        eprintln!("Failed to decode program ID: {}", e);
+        Vec::new()
+    });
+
+    let pk_data = [1; 32];
+    let time_lock = now_sec() - 3600;
+    let taker_pub = coin.key_pair.pubkey().to_string();
+    let taker_pub = Pubkey::from_str(taker_pub.as_str()).unwrap();
+    let secret = [0; 32];
+    let secret_hash = sha256(&secret);
+
+    let args = SendPaymentArgs {
+        time_lock_duration: 0,
+        time_lock,
+        other_pubkey: taker_pub.as_ref(),
+        secret_hash: secret_hash.as_slice(),
+        amount: "0.01".parse().unwrap(),
+        swap_contract_address: &Some(Bytes::from(solana_program_id.clone())),
+        swap_unique_data: &[],
+        payment_instructions: &None,
+        watcher_reward: None,
+        wait_for_confirmation_until: 0,
+    };
+    let tx = coin.send_maker_payment(args).wait().unwrap();
+    println!("swap tx {:?}", tx);
+
+    let refund_args = RefundPaymentArgs {
+        payment_tx: &tx.tx_hex(),
+        time_lock,
+        other_pubkey: taker_pub.as_ref(),
+        tx_type_with_secret_hash: SwapTxTypeWithSecretHash::TakerOrMakerPayment {
+            maker_secret_hash: secret_hash.as_slice(),
+        },
+        swap_contract_address: &Some(Bytes::from(solana_program_id)),
+        swap_unique_data: pk_data.as_slice(),
+        watcher_reward: false,
+    };
+    let refund_tx = block_on(coin.send_maker_refunds_payment(refund_args)).unwrap();
+    println!("refund tx {:?}", refund_tx);
+}
+
+#[test]
+fn solana_coin_send_and_spend_maker_payment() {
+    let passphrase = "federal stay trigger hour exist success game vapor become comfort action phone bright ill target wild nasty crumble dune close rare fabric hen iron".to_string();
+    let (_, coin) = solana_coin_for_test(passphrase, SolanaNet::Devnet);
+    let solana_program_id = "AEwou7VU9bahWRSN2zzgMGqozNY2G27U9fbbdLF9myZx";
+    let solana_program_id = bs58::decode(solana_program_id).into_vec().unwrap_or_else(|e| {
+        eprintln!("Failed to decode program ID: {}", e);
+        Vec::new()
+    });
```

Comment:
Same for using the latest solana program

Reply:
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2091#discussion_r1650835111